tests/test_frame_selection_tools.py

```diff
@@ -0,0 +1,69 @@
+""" Tests for frame selection tools """
+from unittest.mock import Mock
+
+import deeplabcut.utils.frameselectiontools as fst
+
+
+def test_uniform_frames():
+    # TODO: What happens if there are less frames in the video that the number we asked to select?
+    #   -> return all frames from the video?
+
+    # Index bounds: (0, 1) => valid_frame_index_1 == frame_index_1
+    frame_index_1 = list(range(100, 500, 5))
+    # Index bounds: 0.25 * 1600 = 400, 0.3 * 1600 = 480
+    frame_index_2 = list(range(100, 500, 5))
+    valid_frame_index_2 = [idx for idx in frame_index_2 if 400 <= idx <= 480]
+    for fps, duration, n_to_pick, start, end, index, valid_frame_indices in [
+        (32, 10, 10, 0, 1, None, list(range(0, 320))),  # 320 frames at 32 fps -> 10 seconds
+        (16, 100, 50, 0, 1, frame_index_1, frame_index_1),  # 1600 frames at 16 fps -> 100 seconds
+        (16, 100, 5, 0.25, 0.3, frame_index_2, valid_frame_index_2),  # 1600 frames at 16 fps -> 100 seconds
```

Comment:
Can these multiple sets of parameters be moved outside the func in @pytest.mark.parametrize()?; e.g., as in https://github.com/DeepLabCut/DeepLabCut/blob/9fcf1f8ad9f04f2e23da74a3a9dc1463ee831c8f/tests/test_video.py#L76
When a unit test fails, this makes debugging easier.